browser/microsoft_edge_protocol_util.cc

```diff
@@ -0,0 +1,59 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/microsoft_edge_protocol_util.h"
+
+#include "base/strings/string_split.h"
+#include "base/strings/string_util.h"
+#include "base/strings/sys_string_conversions.h"
+#include "base/strings/utf_string_conversions.h"
+#include "url/url_util.h"
+
+namespace {
+
+std::string DecodeURL(const std::string& url) {
+  url::RawCanonOutputT<char16_t> unescaped;
+  url::DecodeURLEscapeSequences(url.data(), url.size(),
+                                url::DecodeURLMode::kUTF8OrIsomorphic,
+                                &unescaped);
+
+  std::string output;
+  base::UTF16ToUTF8(unescaped.data(), unescaped.length(), &output);
+  return output;
+}
+
+}  // namespace
+
+absl::optional<std::string> GetURLFromMSEdgeProtocol(
+    const std::wstring& command_line_arg) {
+  constexpr wchar_t kMSEdgeProtocol[] = L"microsoft-edge:";
```

Comment:
what if we go with ```constexpr base::WStringPiece``` instead? this way we don't have to play with ```wcslen``` and all ```substr``` will become almost free.
also ```SysWideToUTF8``` -> ```WideToUTF8``` in this case.